api/v1alpha1/controlplane_types.go

```diff
@@ -90,12 +107,26 @@ type SecretReference struct {
 	// `name` is the name of the secret.
 	// Required
 	Name string `json:"name"`
-	// Required
+	// +optional
```

Comment:
The reader should be told when this field is present, when it is absent, and how the clients are expected to cope when it is absent or why it will never be a problem when it is absent.

Reply:
ok, added those clarifications.